src/index/vchordrq/am/am_build.rs

```diff
@@ -1287,6 +1339,252 @@ fn make_default_build(
     }]
 }
 
+fn make_internal_approximate_build(
+    vector_options: VectorOptions,
+    internal_build: VchordrqInternalBuildOptions,
+    samples: Vec<Vec<f32>>,
+    reporter: &PostgresReporter,
+) -> (Vec<Structure<Normalized>>, HashMap<usize, usize>) {
+    let mut result = Vec::<Structure<Vec<f32>>>::new();
+    let (top_list, bottom_list) = if internal_build.lists.len() == 1 {
+        let top_k = (internal_build.lists[0] as f64).sqrt().floor() as u32;
+        let top_k = top_k.clamp(1, (samples.len() as f64).sqrt().floor() as u32);
+        (top_k, internal_build.lists[0])
+    } else {
+        (internal_build.lists[0], internal_build.lists[1])
```

Comment:
`lists` should be independent.

Reply:
Fixed with new function `approx_kmeans` which only generate centroids. The recall/QPS is still the same like before.